core/vm/evm.go

```diff
@@ -838,7 +849,43 @@ func (evm *EVM) captureOperationMode(isWriteOperation bool) error {
 	return nil
 }
 
-// enforceMultitenancyCheck returns true if EVM is enforced to do multitenancy check, false otherwise
+// Quorum
+//
+// captureAffectedContract stores the contract address to the affectedContract list if not yet there.
+// The affected mode is also updated if required.
+// Default affected reason is MessageCall.
+// In simulation/eth_call for multitenancy, it sets the expectation of AffectedMode
+// to be verified later when an opcode is executed.
+func (evm *EVM) captureAffectedContract(address common.Address, mode AffectedMode) error {
+	if _, ok := evm.affectedContracts[address]; !ok {
+		evm.affectedContracts[address] = newAffectedType(MessageCall, mode)
+	}
+
+	if !evm.SupportsMultitenancy || evm.AuthorizeMessageCallFunc == nil {
+		return nil
+	}
+	affectedType, found := evm.affectedContracts[address]
```

Comment:
On line 860 you are already looking for `evm.affectedContracts[address]` we probably do not need to do it a second time here.